rp2040-hal/src/pio.rs

```diff
@@ -372,6 +372,50 @@ pub struct UninitStateMachine<SM: ValidStateMachine> {
 unsafe impl<SM: ValidStateMachine + Send> Send for UninitStateMachine<SM> {}
 
 impl<SM: ValidStateMachine> UninitStateMachine<SM> {
+    /// Sets unmasked pin directions
+    ///
+    /// The iterator's item are pairs of `(pin_number, pin_high)`.
+    pub fn set_pins_with_iter(&mut self, pins: impl Iterator<Item = (u8, bool)>) {
```

Comment:
I think `IntoIterator` would be more flexible. Then one could use this as
```rust
sm.set_pins_with_iter([(1, false), (5, true)]);
```

Reply:
Good point, that indeed makes it much nicer to use!